### PR TITLE
CBG-2679 use 3.0 metadata for in memory implementations as well

### PR DESCRIPTION
**Pull request by torcolvin**
Fixes https://jenkins.sgwdev.com/job/MasterIntegration/387/testReport/junit/github/com_couchbase_sync_gateway_base/TestCBGTIndexCreationSafeLegacyName/

I could reproduce with `-count 100`

## Pre-review checklist
- [x] Removed debug logging (`fmt.Print`, `log.Print`, ...)
- [x] Logging sensitive data? Make sure it's tagged (e.g. `base.UD(docID)`, `base.MD(dbName)`)
- [x] Updated relevant information in the API specifications (such as endpoint descriptions, schemas, ...) in `docs/api`

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- [ ] `GSI=true,xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/1528/

### CODE REVIEW

File: base/cbgt.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+)
+
+// NewCbgtCfgMem runs cbgt.NewCfgMem and sets the matching version number we expect for Sync Gateway.
+func NewCbgtCfgMem() (*cbgt.CfgMem, error) {
+	cfg := cbgt.NewCfgMem()
+	cas, err := cfg.Set(cbgt.VERSION_KEY, []byte(SGCbgtMetadataVersion), 0)
+	if err != nil {
+		return nil, err
+	}
+	expectedCas := uint64(1)
```
Comment (adamcfraser) [inline base/cbgt.go]: Why do we care about the CAS value CfgMem assigns to the VERSION_KEY entry?

File: base/cbgt.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import (
+	"fmt"
+
+	"github.com/couchbase/cbgt"
+)
+
+// NewCbgtCfgMem runs cbgt.NewCfgMem and sets the matching version number we expect for Sync Gateway.
+func NewCbgtCfgMem() (*cbgt.CfgMem, error) {
+	cfg := cbgt.NewCfgMem()
+	cas, err := cfg.Set(cbgt.VERSION_KEY, []byte(SGCbgtMetadataVersion), 0)
```
Comment (adamcfraser) [inline base/cbgt.go]: I would expect cbgt to be writing the version key, in the same way it does when running against a document-backed store.  Why do we need to define it in advance in the Cfg key value store when using CfgMem?

**Review by adamcfraser — CHANGES_REQUESTED** [review]

File: base/cbgt.go (hunk 1 above)
Comment (torcolvin) [reply base/cbgt.go]: We don't, I can remove this check.

File: base/cbgt.go (hunk 2 above)
Comment (torcolvin) [reply base/cbgt.go]: The spec of the implmentation is such that it reads the key out of the config if it's not set, so this matches our not in memory version https://github.com/couchbase/sync_gateway/blob/master/base/sg_cluster_cfg.go#L69

Comment (adamcfraser) [reply base/cbgt.go]: Makes sense - thanks.

**Review by adamcfraser — APPROVED** [review]

**Pull request merged**